composer/core/state.py

```diff
@@ -200,6 +200,10 @@ def _create_device_mesh(
     fsdp_config: Optional[Dict[str, Any]],
     tp_config: Optional[Dict[str, Any]],
 ) -> Optional[DeviceMesh]:
+    if version.parse(torch.__version__.split('.dev')[0]) < version.parse('2.3.0'):
+        # Device mesh has correctness issues before torch 2.3.0
```

Comment:
should we raise error if users want to use 2d-parallel with torch version <2.3.0 ?

Reply:
Yep, this happens at in `state._validate_parallelism_configs`